src/main/java/com/getindata/connectors/http/internal/sink/httpclient/JavaNetSinkHttpClient.java

```diff
@@ -110,15 +120,28 @@ private CompletableFuture<List<JavaNetHttpResponseWrapper>> submitRequests(
     }
 
     private SinkHttpClientResponse prepareSinkHttpClientResponse(
-            List<JavaNetHttpResponseWrapper> responses) {
+            List<JavaNetHttpResponseWrapper> responses,
+            String endpointUrl) {
         var successfulResponses = new ArrayList<HttpSinkRequestEntry>();
         var failedResponses = new ArrayList<HttpSinkRequestEntry>();
 
         for (var response : responses) {
             var sinkRequestEntry = response.getSinkRequestEntry();
+            var optResponse = response.getResponse();
+
+            boolean isFailed;
+            if (optResponse.isEmpty()) {
+                isFailed = true;
+            } else {
+                if (httpSinkPostRequestCallback != null) {
+                    httpSinkPostRequestCallback.call(
+                        optResponse.get(), sinkRequestEntry, endpointUrl, headerMap);
+                }
+                isFailed = statusCodeChecker.isErrorCode(optResponse.get().statusCode());
```

Comment:
previously the condition was 
`response.getResponse().isEmpty() || statusCodeChecker.isErrorCode(...), `

now `response.getResponse().isEmpty()` condition is removed, why?

Reply:
I mean, no/empty response from endpoint could also be passed to `httpSinkPostRequestCallback` in my opinion it may be a valid thing to note in audit log for example.

Reply:
`null` as response is passed now